### PR TITLE
Finalize sorting implementation

### PR DESCRIPTION
**Pull request by rolandyuwy**
- Add more JUnit tests for ComparatorUtil, Priority, ExpiryDate, Description, UserPrefs
- Add defensive programming code to Priority, ExpiryDate and Description
- Add more logging

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -8,43 +8,54 @@ title: User Guide
...
-* [5. FAQs](#5-faq)
-* [6. Command summary](#6-command-summary)
-* [7. Glossary](#7-glossary)
+* [4. Application Layout](#4-application-layout)
+* [5. Features](#5-features)
+  * [5.1. Basic Commands](#51-basic-commands)
+    * [5.1.1. Adding a food item](#511-adding-a-food-item)
+    * [5.1.2. Editing a food item](#512-editing-a-food-item)
+    * [5.1.3. Deleting a food item](#513-deleting-a-food-item)
+    * [5.1.5. Changing the quantity of a food item](#514-changing-the-quantity-of-a-food-item)
```
Comment (jonahtanjz) [inline docs/UserGuide.md]: Should be 5.1.4

**Review by jonahtanjz — COMMENTED** [review]
One minor issue. Else LGTM

File: docs/UserGuide.md [hunk 2]
```diff
@@ -8,43 +8,54 @@ title: User Guide
...
+    * [5.3.2. Finding food items](#532-finding-food-items)
+    * [5.3.3. Viewing expired food items](#533-viewing-expired-food-items)
+  * [5.4. Miscellaneous Commands](#54-miscellaneous-commands)
+    * [5.4.1. Undoing previous command](#541-undoing-previous-command)
+    * [5.4.2. Redoing previously undone command](#542-redoing-previously-undone-command)     
+    * [5.4.3. Clearing all entries](#543-clearing-all-entries)
+    * [5.4.4. Viewing help](#544-viewing-help)
+    * [5.4.5. Exiting the program](#545-exiting-the-program)
+  * [5.5. Saving the data](#55-saving-the-data)
+* [6. Upcoming Features](#6-upcoming-features)
```
Comment (jonahtanjz) [inline docs/UserGuide.md]: Update link to match full header. Currently does not go to section when clicked

File: docs/UserGuide.md (hunk 1 above)
Comment (rolandyuwy) [reply docs/UserGuide.md]: Thanks!

File: docs/UserGuide.md (hunk 2 above)
Comment (rolandyuwy) [reply docs/UserGuide.md]: Thanks!

**Review by songyiang — APPROVED** [review]
LGTM

**Pull request merged**